### ISSUE TITLE
Image to text example doesn't work as expected

### ISSUE THREAD

**Issue by anphu7492** [OpenNMT/OpenNMT-py#888]
I ran the demo with the latest code but the results were completely off. The training accuracy skyrocketed to nearly 100% after 3k steps (validation accuracy was around 40%) and then dropped rapidly after 30k steps.

**Comment by da03** [comment]
Hmm are you using the toy dataset included in the demo?

**Comment by anphu7492** [comment]
yes, I followed the instruction. I think 100k steps for training is too much but I also ran the translation at 5k step, the results were also bad.

**Comment by da03** [comment]
No problem! Closing this issue for now, please reopen it if the problem remains.

**Comment by anphu7492** [comment]
Hi, so I trained with large dataset and it worked. However, I still needed to make some changes to the code. Below are the changes that I've made, I can create a pull request if you want. 
```
@@ -187,7 +187,7 @@ class ImageDataset(DatasetBase):
         """
         fields = {}

-        def make_img(data, vocab):
+        def make_img(data, vocab, is_training):
             """ ? """
             # c = data[0].size(0)
             c = max([t.size(0) for t in data])
@@ -200,7 +200,7 @@ class ImageDataset(DatasetBase):
             return imgs

         fields["src"] = torchtext.data.Field(
-            use_vocab=False, dtype=torch.float,
+            use_vocab=False, tensor_type=torch.FloatTensor,
             postprocessing=make_img, sequential=False)

@@ -246,7 +246,7 @@ class ImageDataset(DatasetBase):
             return alignment

         fields["src_map"] = torchtext.data.Field(
-            use_vocab=False, dtype=torch.float,
+            use_vocab=False, tensor_type=torch.FloatTensor,
             postprocessing=make_src, sequential=False)

         def make_tgt(data, vocab):
@@ -258,11 +258,11 @@ class ImageDataset(DatasetBase):
             return alignment

         fields["alignment"] = torchtext.data.Field(
-            use_vocab=False, dtype=torch.long,
+            use_vocab=False, tensor_type=torch.LongTensor,
             postprocessing=make_tgt, sequential=False)

         fields["indices"] = torchtext.data.Field(
-            use_vocab=False, dtype=torch.long,
+            use_vocab=False, tensor_type=torch.LongTensor,
             sequential=False)
```

**Comment by da03** [comment]
Hmm I think you are using a lower torchtext version?

**Comment by anphu7492** [comment]
ok you're right, I downgraded torchtext earlier today to test previous versions.
